code/modules/mob/dead/new_player/new_player.dm

```diff
@@ -431,46 +431,81 @@
 		if(prioritized_job.current_positions >= prioritized_job.total_positions)
 			SSjob.prioritized_jobs -= prioritized_job
 
-	if(length(SSjob.prioritized_jobs))
-		dat += "<div class='notice red'>The station has flagged these jobs as high priority:<br>"
-		var/amt = length(SSjob.prioritized_jobs)
-		var/amt_count
-		for(var/datum/job/a in SSjob.prioritized_jobs)
-			amt_count++
-			if(amt_count != amt) // checks for the last job added.
-				dat += " [a.title], "
-			else
-				dat += " [a.title]. </div>"
+	if(!available_job_count)
+		dat += "<div class='notice red'>There are currently no open positions!</div>"
 
-	dat += "<div class='clearBoth'>Choose from the following open positions:</div><br>"
-	dat += "<div class='jobs'><div class='jobsColumn'>"
-	var/job_count = 0
-	for(var/datum/job/job in SSjob.occupations)
-		if(job && IsJobUnavailable(job.title, TRUE) == JOB_AVAILABLE)
-			job_count++;
-			if (job_count > round(available_job_count / 2))
-				dat += "</div><div class='jobsColumn'>"
-			var/position_class = "otherPosition"
-			if (job.title in GLOB.command_positions)
-				position_class = "commandPosition"
-			dat += "<a class='[position_class]' href='byond://?src=[REF(src)];SelectedJob=[job.title]'>[job.title] ([job.current_positions])</a><br>"
-	if(!job_count) //if there's nowhere to go, overflow opens up.
+	else
+
+	// if(length(SSjob.prioritized_jobs))
+	// 	dat += "<div class='notice red'>The station has flagged these jobs as high priority:<br>"
+	// 	for(var/datum/job/a in SSjob.prioritized_jobs)
+	// 		dat += " [a.title], "
+	// 	dat += "</div>"
+
+		dat += "<div class='clearBoth'>Choose from the following open positions:</div><br>"
+		var/list/categorizedJobs = list(
+			"Command" = list(jobs = list(), titles = GLOB.command_positions, color = "#aac1ee"),
+			"Engineering" = list(jobs = list(), titles = GLOB.engineering_positions, color = "#ffd699"),
+			"Supply" = list(jobs = list(), titles = GLOB.supply_positions, color = "#ead4ae"),
+			"Miscellaneous" = list(jobs = list(), titles = list(), color = "#ffffff", colBreak = 1),
+			"Synthetic" = list(jobs = list(), titles = GLOB.nonhuman_positions, color = "#ccffcc"),
+			"Service" = list(jobs = list(), titles = GLOB.civilian_positions, color = "#cccccc"),
+			"Medical" = list(jobs = list(), titles = GLOB.medical_positions, color = "#99ffe6", colBreak = 1),
+			"Science" = list(jobs = list(), titles = GLOB.science_positions, color = "#e6b3e6"),
+			"Security" = list(jobs = list(), titles = GLOB.security_positions, color = "#ff9999"),
+		)
 		for(var/datum/job/job in SSjob.occupations)
-			if(job.title != SSjob.overflow_role)
+			if(job && IsJobUnavailable(job.title, TRUE) == JOB_AVAILABLE)

```

Comment:
![image](https://user-images.githubusercontent.com/20558591/58579448-5c0c9f80-824b-11e9-8590-da0b9af80c81.png)

Reply:
Ill speak to Jordie on what he fully means by this

Reply:
What do you need clarified?

Reply:
Retrieving `SSjob.occupations.<job>.current_positions`, Im not used to TGs job SS yet so it will take me a while to figure out things

Reply:
SSjob has lists of all existing occupations and their datums https://github.com/tgstation/tgstation/blob/master/code/controllers/subsystem/job.dm#L59-L61
So instead of building a list here of the datums just to read their `current_positions` you can directly get it from SSjob with something like
`SSjob.name_occupations["Janitor"].current_positions`

Reply:
So I need to do one of these for each job? Or loop through a set list of all jobs and substitute that in?

I dont know which of these ideas would be better, so apologies for the fuckload of questions

Reply:
@Jordie0608

Reply:
Sorry for the quite late response; I figured it would just be easier to show than tell so I've made you a replacement proc:

```
/mob/dead/new_player/proc/LateChoices()
	var/dat = "<div class='notice'>Round Duration: [DisplayTimeText(world.time - SSticker.round_start_time)]</div>"
	if(SSshuttle.emergency)
		switch(SSshuttle.emergency.mode)
			if(SHUTTLE_ESCAPE)
				dat += "<div class='notice red'>The station has been evacuated.</div><br>"
			if(SHUTTLE_CALL)
				if(!SSshuttle.canRecall())
					dat += "<div class='notice red'>The station is currently undergoing evacuation procedures.</div><br>"
	var/any_jobs_available = FALSE
	for(var/datum/job/job in SSjob.occupations)
		if(job && IsJobUnavailable(job.title, TRUE) == JOB_AVAILABLE)
			any_jobs_available = TRUE
			break
	if(!any_jobs_available)
		dat += "<div class='notice red'>There are currently no open positions!</div>"
	else
		for(var/datum/job/prioritized_job in SSjob.prioritized_jobs)
			if(prioritized_job.current_positions >= prioritized_job.total_positions)
				SSjob.prioritized_jobs -= prioritized_job
		dat += "<table><tr><td valign='top'>"
		for(var/category in GLOB.command_positions + GLOB.engineering_positions + GLOB.supply_positions + GLOB.nonhuman_positions + GLOB.civilian_positions + GLOB.medical_positions + GLOB.science_positions + GLOB.security_positions)
			var/cat_color = SSjob.name_occupations[job].selection_color //use the color of the first job in the category (the department head) as the category color
			dat += "<fieldset style='border: 2px solid [cat_color]; display: inline'>"
			dat += "<legend align='center' style='color: [cat_color]'>[jobcat]</legend>"
			for(var/job in category)
				var/datum/job/job_datum = SSjob.name_occupations[job]
				if(job_datum in SSjob.prioritized_jobs)
					dat += "<[job in GLOB.command_positions ? "class='bold' " : ""]style='display:block;width:170px' href='byond://?src=[REF(src)];SelectedJob=[job_datum.title]'><class='greenannounce'>[job_datum.title] ([job_datum.current_positions])</class></a>"
				else
					dat += "<[job in GLOB.command_positions ? "class='bold' " : ""]style='display:block;width:170px' href='byond://?src=[REF(src)];SelectedJob=[job_datum.title]'>[job_datum.title] ([job_datum.current_positions])</a>"
			dat += "</fieldset><br>"
		dat += "</td></tr></table></center>"
		dat += "</div></div>"
	var/datum/browser/popup = new(src, "latechoices", "Choose Profession", 680, 580)
	popup.set_content(dat)
	popup.open(FALSE) // 0 is passed to open so that it doesn't use the onclose() proc
```

It's quite rough, I didn't even check if it compiled, so you'll probably need to do some editing to make it work.
Note on the use of `selection_color`, the only other place this is used is the character setup occupation preferences so the colors may be a bit different but it'd be good to have that and the latejoin use the same ones, feel free to change both to suit.
I also dropped the use of `var/position_class = "otherPosition"` as a css selector for some existing css classes we have, this means you can delete `html/browser/playeroptions.css` entirely as it'll be unused.

If you have questions ping me in discord, I'm more likely to respond within a week that way =P

Reply:
@Jordie0608 I tried this code, made numerous edits but couldnt get it to smart-pick the category colours, constantly telling me the list was out of bounds. Fir it tried to pull from `job` which isnt defined in this line
`var/cat_color = SSjob.name_occupations[job].selection_color `

I then treid to pull `category[1]` but it kept throwing an out of bounds at me.

Reply:
Yeah I missed quite a few things. When I'm home i'll try actually testing this and message you with an update.